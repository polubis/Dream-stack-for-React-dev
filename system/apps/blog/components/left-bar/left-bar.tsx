import {
  ArrowTopIcon,
  Button,
  DiscordIcon,
  LinkedinIcon,
  LeftBar as UILeftBar,
  UserIcon,
} from '@system/figa-ui';
import {
  GREEN_ON_SOFTWARE_AUTHOR,
  GREEN_ON_SOFTWARE_DISCORD,
  GREEN_ON_SOFTWARE_LINKEDIN,
} from 'apps/blog/consts';

const LeftBar = () => {
  return (
    <UILeftBar>
      <Button shape="rounded" size={2}>
        <ArrowTopIcon />
      </Button>
      <a
        href={GREEN_ON_SOFTWARE_DISCORD}
        title="GreenOn Software Discord channel"
        target="_blank"
        rel="noreferrer"
      >
        <Button shape="rounded" size={2}>
          <DiscordIcon />
        </Button>
      </a>
      <a
        href={GREEN_ON_SOFTWARE_AUTHOR}
        title="Adrian Połubiński Linkedin"
        target="_blank"
        rel="noreferrer"
      >
        <Button shape="rounded" size={2}>
          <UserIcon />
        </Button>
      </a>
      <a
        href={GREEN_ON_SOFTWARE_LINKEDIN}
        title="Linkedin GreenOn Software profile"
        target="_blank"
        rel="noreferrer"
      >
        <Button shape="rounded" size={2}>
          <LinkedinIcon />
        </Button>
      </a>
    </UILeftBar>
  );
};

export { LeftBar };
