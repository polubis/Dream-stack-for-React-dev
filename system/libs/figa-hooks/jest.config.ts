/* eslint-disable */
export default {
  displayName: 'figa-hooks',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/figa-hooks',
};
