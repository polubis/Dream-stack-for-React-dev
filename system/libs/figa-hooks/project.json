{
  "name": "figa-hooks",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/figa-hooks/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/figa-hooks/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nrwl/vite:build",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/libs/figa-hooks"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/figa-hooks/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
