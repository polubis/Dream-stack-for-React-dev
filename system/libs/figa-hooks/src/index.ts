export * from './lib/use-portal';
export * from './lib/use-toggle';
export * from './lib/use-intersection-observer';
export * from './lib/use-click-outside';
export * from './lib/use-stepper';
export * from './lib/use-scroll-y';
export * from './lib/use-client-effect';
export * from './lib/use-client-layout-effect';
export * from './lib/use-element-size';
export * from './lib/use-clipboard';
export * from './lib/use-popover';
