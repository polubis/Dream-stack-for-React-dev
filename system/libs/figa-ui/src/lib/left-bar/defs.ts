import type { ReactNode } from 'react';

interface LeftBarProps {
  className?: string;
  children: ReactNode;
}

export type { LeftBarProps };
