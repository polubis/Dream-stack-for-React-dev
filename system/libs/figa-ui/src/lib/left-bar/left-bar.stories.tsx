import type { Story, Meta } from '@storybook/react';

import { LeftBar } from './left-bar';
import { Button } from '../button';

export default {
  component: LeftBar,
  title: 'LeftBar',
} as Meta;

const Template: Story = () => {
  return (
    <div style={{ minHeight: '150vh' }}>
      <LeftBar>
        <Button size={2}>A</Button>
        <Button size={2}>B</Button>
        <Button size={2}>C</Button>
      </LeftBar>
    </div>
  );
};

export const Default = Template.bind({});
Default.args = {};
