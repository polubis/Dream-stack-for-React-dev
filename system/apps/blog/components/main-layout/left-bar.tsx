import { useScrollTo } from '@system/figa-hooks';
import {
  ArrowTopIcon,
  Bar,
  Button,
  HalfMoonIcon,
  SunIcon,
  useThemeProvider,
} from '@system/figa-ui';

const LeftBar = () => {
  const theme = useThemeProvider();
  const [, { toTop }] = useScrollTo();

  return null

  // return (
  //   <Bar>
  //     <Button
  //       size={2}
  //       shape="rounded"
  //       onClick={() => theme.setTheme(theme.key === 'dark' ? 'light' : 'dark')}
  //     >
  //       {theme.key === 'dark' ? <SunIcon /> : <HalfMoonIcon />}
  //     </Button>
  //     <Button shape="rounded" size={2} onClick={() => toTop()}>
  //       <ArrowTopIcon />
  //     </Button>
  //   </Bar>
  // );
};

export { LeftBar };
