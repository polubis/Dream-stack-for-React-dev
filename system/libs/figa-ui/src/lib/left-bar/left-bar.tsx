import { useScroll } from '@system/figa-hooks';

import c from 'classnames';
import type { LeftBarProps } from './defs';

const LeftBar = ({ children }: LeftBarProps) => {
  const [{ is }] = useScroll({ delay: 0 });

  return (
    <div
      className={c(
        'left-bar',
        is === 'regress' || is === 'idle' ? 'visible' : 'hidden'
      )}
    >
      {children}
    </div>
  );
};

export { LeftBar };
