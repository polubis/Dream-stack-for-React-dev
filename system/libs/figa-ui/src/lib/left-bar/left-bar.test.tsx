import { render, screen } from '@testing-library/react';
import { useScrollY, type UseScrollYReturn } from '@system/figa-hooks';

import { LeftBar } from './left-bar';

jest.mock('@system/figa-hooks', () => ({
  useScrollY: jest.fn(),
}));

describe('Left bar can be used when', () => {
  const leftBarFixture = (useScrollYReturn: Omit<UseScrollYReturn, 'ref'>) => {
    (useScrollY as jest.Mock).mockImplementationOnce(() => useScrollYReturn);

    const result = render(
      <LeftBar className="my-class">
        <div>A</div>
        <div>B</div>
      </LeftBar>
    );

    return result;
  };

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = leftBarFixture({
      state: {
        direction: 'idle',
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] bar is visible when nothing happened', () => {
    const { container } = leftBarFixture({
      state: {
        direction: 'idle',
      },
    });

    screen.getByText('A');
    screen.getByText('B');

    expect(
      (container.querySelector('.left-bar') as HTMLElement).className
    ).toBe('left-bar visible');
  });

  it('[FRAGILE] bar is hidden when scrolling down', () => {
    const { container } = leftBarFixture({
      state: {
        direction: 'down',
        previousScroll: 0,
        currentScroll: 100,
      },
    });

    screen.getByText('A');
    screen.getByText('B');

    expect(
      (container.querySelector('.left-bar') as HTMLElement).className
    ).toBe('left-bar hidden');
  });

  it('[FRAGILE] bar is visible when scrolling up', () => {
    const { container } = leftBarFixture({
      state: {
        direction: 'up',
        previousScroll: 0,
        currentScroll: 100,
      },
    });

    screen.getByText('A');
    screen.getByText('B');

    expect(
      (container.querySelector('.left-bar') as HTMLElement).className
    ).toBe('left-bar visible');
  });
});
