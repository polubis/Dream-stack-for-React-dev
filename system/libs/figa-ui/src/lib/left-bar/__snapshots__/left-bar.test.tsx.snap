// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left bar can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="left-bar visible"
  >
    <div>
      A
    </div>
    <div>
      B
    </div>
  </div>
</DocumentFragment>
`;
